src/ArchGDAL.jl

```diff
@@ -12,6 +12,9 @@ using CEnum
 
 const GFT = GeoFormatTypes
 
+const gdal_vector_drivers = ["AmigoCloud", "Arrow", "AVCBIN", "AVCE00", "CAD", "CARTO", "CSV", "CSW", "DGN", "DGNv8", "DWG", "DXF", "EDIGEO", "EEDA", "Elasticsearch", "ESRIJSON", "FileGDB", "FlatGeobuf", "Geoconcept", "GeoJSON", "GeoJSONSeq", "GeoRSS", "GML", "GMLAS", "GMT", "GPKG", "GPSBabel", "GPX", "GRASS", "HANA", "IDB", "IDRISI", "INTERLIS 1", "INTERLIS 2", "JML", "KML", "LIBKML", "LVBAG", "MapML", "Memory", "MapInfo File", "MongoDBv3", "MSSQLSpatial", "MVT", "MySQL", "NAS", "netCDF", "NGW", "UK .NTF", "OAPIF", "OCI", "ODBC", "ODS", "OGDI", "OpenFileGDB", "OSM", "Parquet", "PDF", "PDS", "PostgreSQL", "PGDump", "PGeo", "PLScenes", "S57", "SDTS", "Selafin", "ESRI Shapefile", "SOSI", "SQLite", "SVG", "SXF", "TIGER", "TopoJSON", "VDV", "VFK", "WAsP", "WFS", "XLS", "XLSX"]
+const gdal_raster_drivers = ["AAIGrid", "ACE2", "ADRG", "AIG", "AIRSAR", "ARG", "BAG", "BASISU", "BLX", "BMP", "BSB", "BT", "BYN", "CAD", "CALS", "CEOS", "COASP", "COG", "COSAR", "CPG", "CTable2", "CTG", "DAAS", "DDS", "DERIVED", "DIMAP", "DIPEx", "DOQ1", "DOQ2", "DTED", "ECRGTOC", "ECW", "EEDAI", "EHdr", "EIR", "ELAS", "ENVI", "ERS", "ESAT", "ESRIC", "EXR", "FAST", "FIT", "FITS", "GenBin", "GeoRaster", "GFF", "GIF", "GPKG", "GRASS", "GRASSASCIIGrid", "GRIB", "GS7BG", "GSAG", "GSBG", "GSC", "GTA", "GTiff", "GXF", "HDF4", "HDF5", "HEIF", "HF2", "HFA", "RST", "ILWIS", "IRIS", "ISCE", "ISG", "ISIS2", "ISIS3", "JDEM", "JP2ECW", "JP2KAK", "JP2LURA", "JP2MrSID", "JP2OpenJPEG", "JPEG", "JPEGXL", "JPIPKAK", "KEA", "KMLSuperoverlay", "KRO", "KTX2", "L1B", "LAN", "LCP", "Leveller", "LOSLAS", "MAP", "MRF", "MBTiles", "MEM", "MFF", "MFF2", "MrSID", "MSG", "MSGN", "NDF", "netCDF", "NGSGEOID", "NGW", "NITF", "NTv2", "NWT_GRD", "NWT_GRC", "OGCAPI", "OZI", "JAXAPALSAR", "PAux", "PCIDSK", "PCRaster", "PDF", "PDS", "PDS4", "PLMosaic", "PNG", "PNM", "PostGISRaster", "PRF", "R", "Rasdaman", "Rasterlite", "SQLite", "RDB", "RIK", "RMF", "ROI_PAC", "RPFTOC", "RRASTER", "RS2", "SAFE", "SAR_CEOS", "SAGA", "SDTS", "SENTINEL2", "SGI", "SIGDEM", "SNODAS", "SRP", "SRTMHGT", "STACIT", "STACTA", "Terragen", "TGA", "TIL", "TileDB", "TSX", "USGSDEM", "VICAR", "VRT", "WCS", "WEBP", "WMS", "WMTS", "XPM", "XYZ", "Zarr", "ZMAP"]
```

Comment:
1. Unfortunately this feels like a step back from the vision in https://trac.osgeo.org/gdal/wiki/rfc46_gdal_ogr_unification and will be hard to track with future versions of GDAL.

2. Might there be drivers that appear in both vectors?

Reply:
1. Yes. It might be better to implement the approach you suggested below.
2. Yes, there are. E.g. PDF. So this is ambiguous and not a good solution.